seeder/seed.js

```diff
@@ -37,4 +42,148 @@ async function seed() {
   }
 }
 
+async function usersSeed() {
+  let transaction;
+
+  try {
+    const hasUsers = await db.User.findAll();
+
+    if (hasUsers.length > 0) {
+      console.log("Database already seeded with users. Skipping seeding process.");
+      return;
+    }
+
+    transaction = await db.sequelize.transaction();
+
+    for (let i = 0; i < dummyUsers.length; i++) {
+      const { salt, hashedPassword } = await hashPassword(dummyUsers[i].password);
+
+      const user = await db.User.create(
+        {
+          ...dummyUsers[i],
+          username: dummyUsers[i].username,
+          email: dummyUsers[i].email,
+          salt: salt,
+          hashedPassword: hashedPassword,
+        },
+        { transaction }
+      );
+
+      if (!user) {
+        throw new Error("Failed to create user.");
+      }
+    }
+
+    await transaction.commit();
+    console.log("Users seeded successfully.");
+  } catch (error) {
+    console.error("Error seeding users:", error);
+    if (transaction && !transaction.finished) {
+      await transaction.rollback();
+    }
+  }
+}
+
+async function projectsSeed() {
+  let transaction;
+  try {
+    const hasProject = await db.Project.findAll();
+    const users = await db.User.findAll();
+
+    if (hasProject.length > 0) {
+      console.log("Database already seeded with projects. Skipping seeding process.");
+      return;
+    }
+
+    transaction = await db.sequelize.transaction();
+
+    for (let i = 0; i < userProjects.length; i++) {
+      const userId = users[userProjects[i].userId - 1].id;
+      const project = await db.Project.create(
+        {
+          ...userProjects[i],
+          userId: userId,
+        },
+        { transaction }
+      );
+
+      if (!project) {
+        throw new Error("Failed to create project.");
+      }
```

Comment:
When I added some start and end date your types where correctly being applied you can use the setTypes helper function added to the project output from sequelize to do it here.
```suggestion
      const project = await db.Project.create(
        {
          ...userProjects[i],
          userId: userId,
        },
        { transaction }
      );

      if (!project) {
        throw new Error("Failed to create project.");
      }
      await project.setTypes(userProjects[i].types, { transaction });
```